### ISSUE TITLE
swingset-core-eval should not be subject to crank limit

### ISSUE THREAD

**Issue by dckc** [Agoric/Agoric#5047]
## What is the Problem Being Solved?

(raised on behalf of @dtribble . IOU detail)

**Comment by Tartuffo** [comment]
We should not fail to do an upgrade, or other enforcement of Gov, due to metering constraints.  
Could say bootstrap vat is not metered at all.  And maybe give that affordance to Zoe, too.
Since we test all upgrades before doing them in devnet or mainnet, this is not an MN-1 issue.

**Comment by dckc** [comment]
> Could say bootstrap vat is not metered at all. And maybe give that affordance to Zoe, too.

I doubt that would help much. I think the expensive part is evaluating the source code of newly instantiated contracts, which happens in the new vat for that contract, not in the bootstrap/zoe vats.

See also
 - #2268

**Comment by warner** [comment]
#5040 changed the way we launch `xsnap` workers to match the original intention: unmetered vats have neither a per-crank/per-delivery computron limit, nor a cumulative/decrementing/refillable `Meter` -based limit. All static vats are unmetered, and the new `bootstrap.js` code [uses no options](https://github.com/Agoric/agoric-sdk/blob/8e33f53b5a53c456637055acb8ba5c691fd18deb/packages/vats/src/core/basic-behaviors.js#L69) when creating the initial vats, so all initial vats (e.g. `zoe`) are unmetered too. And Zoe [does not use any options](https://github.com/Agoric/agoric-sdk/blob/8e33f53b5a53c456637055acb8ba5c691fd18deb/packages/zoe/src/zoeService/createZCFVat.js#L29) when calling `E(vatAdminSvc).createVat(bundleCap)`, so all contract vats are currently unmetered as well.

On the good side, that means nothing in *any* of the chain-side code is subject to a computron limit, including `bootstrap.js` startup and the "Big JS Hammer" governance mechanism (`eval(proposal)` with access to all bootstrap authority), which I think means this ticket could be closed. On the bad side, it means we're competely vulnerable to infinite loops in all MN-1 vat or upgrade code. (On the other hand, we'd be so surprised for an MN-1 vat to be killed by the compute meter that we'd probably prefer to halt the chain anyways, and it's easier to recover from a hung chain than from a consensus-based vat termination).

@dtribble do you remember what you were thinking of when you asked for this ticket to be created?

**Comment by warner** [comment]
The deeper issue, really, is for the MN-3 timeframe, when most contract vats *are* metered. In that world, I think this ticket says that the bootstrap vat (and any Big JS Hammer activity therein) should remain unmetered. It might also suggest that activity within normally-metered vats should be unmetered if initiated by the Big JS Hammer. We've talked about per-message priorities being connected to metering limits, whereby you could exceed your usual meter if you received a deliberately-empowering message from an unmetered sender.

**Comment by dckc** [comment]
> @dtribble do you remember what you were thinking of when you asked for this ticket to be created?
> ...
> The deeper issue, really, is for the MN-3 timeframe, when most contract vats _are_ metered.

Yes, I think that time when most contract vats _are_ metered is the reason he asked for this issue.
